### ISSUE TITLE
Compilation errors

### ISSUE THREAD

**Issue by Compro-Prasad** [CoatiSoftware/Sourcetrail#788]
Did:
```sh
mkdir build
cd build
cmake ..
make -j2
```
Got:
```
17:45 : build $ make -j2
check/update version number
[  1%] Built target Sourcetrail_lib_utility
-- Found Git: /usr/bin/git (found version "2.24.0") 
-- Version: 2020.1.19
[  1%] Built target versionnumber
[ 53%] Built target Sourcetrail_lib
[ 53%] Automatic MOC for target Sourcetrail_lib_gui
[ 53%] Built target Sourcetrail_lib_gui_autogen
[ 89%] Built target Sourcetrail_lib_gui
[ 90%] Linking CXX executable app/sourcetrail_indexer
[ 91%] Linking CXX executable test/Sourcetrail_test
/usr/bin/ld: libSourcetrail_lib.a(sqlite3.c.o): undefined reference to symbol 'pthread_join@@GLIBC_2.2.5'
/usr/bin/ld: /usr/lib/libpthread.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
/usr/bin/ld: libSourcetrail_lib.a(sqlite3.c.o): undefined reference to symbol 'pthread_join@@GLIBC_2.2.5'
/usr/bin/ld: /usr/lib/libpthread.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/Sourcetrail_indexer.dir/build.make:96: app/sourcetrail_indexer] Error 1
make[1]: *** [CMakeFiles/Makefile2:227: CMakeFiles/Sourcetrail_indexer.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
make[2]: *** [CMakeFiles/Sourcetrail_test.dir/build.make:621: test/Sourcetrail_test] Error 1
make[1]: *** [CMakeFiles/Makefile2:193: CMakeFiles/Sourcetrail_test.dir/all] Error 2
make: *** [Makefile:152: all] Error 2
```

**Comment by mlangkabel** [comment]
Hey, what's your operating system?

**Comment by egraether** [comment]
Can you please test whether pull request #784 fixes this?

**Comment by Compro-Prasad** [comment]
I am on Arch Linux.

**Comment by Compro-Prasad** [comment]
@egraether That PR seems to fix the issue.

**Comment by egraether** [comment]
Fixed by merge of PR #827.
